### PR TITLE
Typescript TCG & Tabletop's plugin

### PR DESCRIPTION
**Pull request by mia-pi-git**
i'm a bit concerned about the `search: any` specification on line 16 but that was the only way to fix a specific error (and two deriving from it)  - lmk if there's a better way to handle that.
Passes tests, though, but was advised against `any` unless there was nothing better (not sure if there is or not, since Object and AnyObject also disagree with it).

### CODE REVIEW

**Comment by Zarel** [comment]
What's wrong with `AnyObject`?

**Comment by mia-pi-git** [comment]
> What's wrong with `AnyObject`?

Last time I attempted it, it threw tsc errors. However, I seem to have fixed whatever was causing the `cannot be used as an index type` error, as it's passing checks now. Fixing one other thing that's running off, then I'll commit.

File: server/chat-plugins/tcgtabletop.ts [hunk 1]
```diff
@@ -38,39 +39,46 @@ async function getFandom(site, pathName, search) {
...
 		if (!this.canBroadcast()) return;
 		if (room.roomid !== 'tcgtabletop') return this.errorReply("This command can only be used in the TCG & Tabletop room.");
-		let subdomain = 'yugioh';
-		let query = target.trim();
+		const subdomain = 'yugioh';
+		const query = target.trim();
 		if (!query) return this.parse('/help yugioh');
 
-		return searchFandom(subdomain, query).then(data => {
+		return searchFandom(subdomain, query).then((data: { url: any, title: any, id: any}) => {
```
Comment (Zarel) [inline server/chat-plugins/tcgtabletop.ts]: I'm surprised this isn't a style error, but there should be no space in front of `url`.

Also, these can all be `unknown`.

```suggestion
		return searchFandom(subdomain, query).then((data: {url: unknown, title: unknown, id: unknown}) => {
```

File: server/chat-plugins/tcgtabletop.ts [hunk 2]
```diff
@@ -38,39 +39,46 @@ async function getFandom(site, pathName, search) {
...
-			let entryTitle = Dex.getString(data.title);
-			let id = Dex.getString(data.id);
+			const entryUrl = Dex.getString(data.url);
+			const entryTitle = Dex.getString(data.title);
+			const id = Dex.getString(data.id);
 			let htmlReply = Chat.html`<strong>Best result for ${query}:</strong><br /><a href="${entryUrl}">${entryTitle}</a>`;
 			if (id) {
-				getCardDetails(subdomain, id).then(card => {
-					let thumb = Dex.getString(card.thumbnail);
+				getCardDetails(subdomain, id).then((card: { thumbnail: any }) => {
```
Comment (Zarel) [inline server/chat-plugins/tcgtabletop.ts]:
```suggestion
				getCardDetails(subdomain, id).then((card: {thumbnail: unknown}) => {
```

File: server/chat-plugins/tcgtabletop.ts [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
 const SEARCH_PATH = '/api/v1/Search/List/';
 const DETAILS_PATH = '/api/v1/Articles/Details/';
 
-async function getFandom(site, pathName, search) {
+async function getFandom(site: string, pathName: string, search: AnyObject) {
 	const reqOpts = {
 		hostname: `${site}.fandom.com`,
 		method: 'GET',
-		path: `${pathName}?${querystring.stringify(search)}`,
+		path: pathName + '?' + querystring.stringify(search),
```
Comment (Zarel) [inline server/chat-plugins/tcgtabletop.ts]:
```suggestion
		path: `${pathName}?${querystring.stringify(search)}`,
```

Can you revert this? I think I prefer how it was before.

**Comment by Zarel** [comment]
@Asheviere tagging Astrid in case she has some opinions. This looks good besides my last suggestion.

File: server/chat-plugins/tcgtabletop.ts [hunk 4]
```diff
@@ -84,7 +92,7 @@ exports.commands = {
 				if (!this.broadcasting) return this.sendReply(`|raw|<div class="infobox">${htmlReply}</div>`);
 				room.addRaw(`<div class="infobox">${htmlReply}</div>`).update();
 			}
-		}, err => {
+		}, (err: AnyObject) => {
```
Comment (Zarel) [inline server/chat-plugins/tcgtabletop.ts]:
```suggestion
		}, (err: Error) => {
```

Does this work?

Comment (mia-pi-git) [reply server/chat-plugins/tcgtabletop.ts]: > does this work

Looks like no: 
> Property 'code' does not exist on type 'Error'.

Comment (Zarel) [reply server/chat-plugins/tcgtabletop.ts]: `NodeJS.Error`?

Comment (Zarel) [reply server/chat-plugins/tcgtabletop.ts]: looks like also no.

Comment (Zarel) [reply server/chat-plugins/tcgtabletop.ts]:
```suggestion
		}, (err: Error & {code: number}) => {
```

**Comment by Zarel** [comment]
I've heard no response, and this seems good, so I guess I'm merging this.

**Pull request merged**